Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -206,40 +218,47 @@ private void StopShooting(EntityUid uid, GunComponent gun)
     /// <summary>
     /// Attempts to shoot at the target coordinates. Resets the shot counter after every shot.
     /// </summary>
-    public void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun, EntityCoordinates toCoordinates)
+    public void AttemptShoot(EntityUid? user, EntityUid gunUid, GunComponent gun, EntityCoordinates toCoordinates, bool needUser = true)
     {
         gun.ShootCoordinates = toCoordinates;
-        AttemptShoot(user, gunUid, gun);
+        AttemptShoot(user, gunUid, gun, needUser);
         gun.ShotCounter = 0;
     }
 
-    private void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun)
+    private void AttemptShoot(EntityUid? user, EntityUid gunUid, GunComponent gun, bool needUser = true)
     {
-        if (gun.FireRate <= 0f ||
-            !_actionBlockerSystem.CanUseHeldEntity(user))
+        if (gun.FireRate <= 0f)
             return;
 
+        if (needUser)
+        {
+            if (user == null) return;
+
```

Comment:
This shouldn't be here, the caller should be handling it or worst case make a new method.